src/js/frontend/src/Redux/reducer.js

```diff
@@ -20,16 +24,37 @@ export function rootReducer (state = IMap(), action) {
 
 const defaultState = {
   showAboutScreen: false,
+  entities: {
+    people: {},
+    organisations: {},
+    "government-offices": {},
+  },
 };
 
-export function aboutReducer(state = defaultState, {type}) {
-  let newState = Object.assign({}, state);
-
+function aboutReducer(state = defaultState.showAboutScreen, {type}) {
   switch (type) {
     case actionTypes.TOGGLE_ABOUT:
-      newState.showAboutScreen = !newState.showAboutScreen;
-      return newState;
+      return ! state;
+    default:
+      return state;
+  }
+}
+
+function addEntityReducer(state = defaultState.entities, action) {
+  let {type, entityType, id, data} = action;
+
+  switch (type) {
+    case actionTypes.ADD_ENTITY:
+      if (! state.has(entityType)) {
+        console.error(`Received ADD_ENTITY action with invalid entity type ${entityType}`);
```

Comment:
In general I agree, but if this ever happens it's a real bug which we want to know about.